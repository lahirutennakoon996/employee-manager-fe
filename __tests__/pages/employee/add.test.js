import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';

import Add from "@/pages/employee/add";
import employeeConfig from "@/config/employee.config";
import mockStore from "../../../__mocks__/react-redux.mock";

describe('Add Employee Form', () => {
  const initialState = {
    employees: {
      allEmployees: [],
    }
  };

  const store = mockStore(initialState);
  let component;

  beforeEach(() => {
    const componentRendered = render(<Provider store={store}><Add /></Provider>);
    component = componentRendered.container;
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('render the first name input', () => {
    const firstNameInput = screen.getByRole('textbox', {
      name: /First Name/,
    });
    expect(firstNameInput).toBeInTheDocument();
    expect(firstNameInput).not.toBeDisabled();
    expect(firstNameInput).toHaveValue('');
  })

  it('render the last name input', () => {
    const lastNameInput = screen.getByRole('textbox', {
      name: /Last Name/,
    });
    expect(lastNameInput).toBeInTheDocument();
    expect(lastNameInput).not.toBeDisabled();
    expect(lastNameInput).toHaveValue('');
  })

  it('render the email input', () => {
    const emailInput = screen.getByRole('textbox', {
      name: /Email/,
    });
    expect(emailInput).toBeInTheDocument();
    expect(emailInput).not.toBeDisabled();
    expect(emailInput).toHaveValue('');
  })

  it('render the phone input', () => {
    const phoneInput = screen.getByRole('textbox', {
      name: /Phone/,
    });
    expect(phoneInput).toBeInTheDocument();
    expect(phoneInput).not.toBeDisabled();
    expect(phoneInput).toHaveValue('+94')
  })

  it('render the gender select', () => {
    const genderInput = screen.getByRole('combobox', {
      name: /Gender/,
    });
    expect(genderInput).toBeInTheDocument();
    expect(genderInput).not.toBeDisabled();

    // Test for dropdown options
    Object.values(employeeConfig.gender).forEach((genderVal) => {
      const option = screen.getByRole('option', {
        name: genderVal
      });
      expect(option).toBeInTheDocument();
      expect(option).not.toBeDisabled();
    })
  })

  it('render the add button', async () => {
    const addBtn = screen.getByRole('button', {
      name: /add/i,
    });

    expect(addBtn).toBeInTheDocument();
    expect(addBtn).not.toBeDisabled();
  })
})