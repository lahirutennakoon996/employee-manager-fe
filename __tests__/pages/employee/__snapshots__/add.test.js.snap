// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Employee Form should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="row text-end mb-5 mt-2"
    >
      <a
        href="/employee/list"
      >
        List View
      </a>
    </div>
    <div
      class="row text-center"
    >
      <form
        action="#"
      >
        <div
          class="row my-3 mx-5"
        >
          <label
            class="col-4 col-form-label text-end"
            for="first_name"
          >
            First Name
          </label>
          <div
            class="col-4"
          >
            <input
              class="form-control"
              id="first_name"
              name="first_name"
              value=""
            />
            <div
              class="errorMsg"
            />
          </div>
        </div>
        <div
          class="row my-3 mx-5"
        >
          <label
            class="col-4 col-form-label text-end"
            for="last_name"
          >
            Last Name
          </label>
          <div
            class="col-4"
          >
            <input
              class="form-control"
              id="last_name"
              name="last_name"
              value=""
            />
            <div
              class="errorMsg"
            />
          </div>
        </div>
        <div
          class="row my-3 mx-5"
        >
          <label
            class="col-4 col-form-label text-end"
            for="email"
          >
            Email
          </label>
          <div
            class="col-4"
          >
            <input
              class="form-control"
              id="email"
              name="email"
              value=""
            />
            <div
              class="errorMsg"
            />
          </div>
        </div>
        <div
          class="row my-3 mx-5"
        >
          <label
            class="col-4 col-form-label text-end"
            for="phone"
          >
            Phone
          </label>
          <div
            class="col-4"
          >
            <input
              class="form-control"
              id="phone"
              name="phone"
              value="+94"
            />
            <div
              class="errorMsg"
            />
          </div>
        </div>
        <div
          class="row my-3 mx-5"
        >
          <label
            class="col-4 col-form-label text-end"
            for="gender"
          >
            Gender
          </label>
          <div
            class="col-4"
          >
            <select
              class="form-select"
              id="gender"
              name="gender"
            >
              <option
                value="M"
              >
                M
              </option>
              <option
                value="F"
              >
                F
              </option>
            </select>
            <div
              class="errorMsg"
            />
          </div>
        </div>
        <button
          class="btn btn-primary"
          type="submit"
        >
          Add
        </button>
      </form>
    </div>
  </div>
</div>
`;
